### ISSUE TITLE
ESLint and Prettier auto-fixes are ignored by the build-flow.

### ISSUE THREAD

**Issue by smellyshovel** [maplibre/maplibre-gl-directions#108]
In the build-flow there's currently a step that checks and auto-fixes all the code with eslint and prettier. The issue is that the changes are not commited and pushed by the actions bot and therefore they are ignored. This must be addressed at some point 'cause as it is now, it's a waste of resources.

**Comment by wipfli** [comment]
In GL JS, we use eslint to check if everything is fine and if warnings and errors occur, the workflow fails. Then the user just needs to run something manually like `npm run lint -- --fix`, solve the problem, and commit the change. I prefer this over automatic commits made by the workflow because it is easier to understand for new people, the commits are controlled by a human, and follow-up workflows will run automatically (GitHub has some rules that workflows should not trigger workflows to avoid loops...).

**Comment by smellyshovel** [comment]
@wipfli sound reasonable enough. But what about the Prettier? It only affects formatting, and I really prefer all the codebase to be style in the same manner. There are no guarantees that each contributor will manually run the Prettier script locally (I even frequently forget to do so myself), so I think it's a good idea to run it during the automated build. What do you think?

**Comment by wipfli** [comment]
Yes completely agree. Let's run Prettier in CI and if something is bad, the workflow should fail. We will only merge things with green CI so like this the codebase stays clean...

**Comment by smellyshovel** [comment]
@wipfli so you mean without auto-fixes on the Prettier side? Just the failing build if something's out of place?

**Comment by wipfli** [comment]
Yes
